pkgs/by-name/li/liana/default.nix

```diff
@@ -61,27 +63,29 @@ rustPlatform.buildRustPackage rec {
     systemd
   ];
 
-  sourceRoot = "${src.name}/gui";
+  sourceRoot = if withGui then "${src.name}/gui" else null;
 
-  postInstall = ''
+  postInstall = if withGui then ''
     install -Dm0644 ./ui/static/logos/liana-app-icon.svg $out/share/icons/hicolor/scalable/apps/liana.svg
     wrapProgram $out/bin/liana-gui --prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath runtimeLibs}"
+  '' else ''
+    install -Dm0644 ./contrib/lianad_config_example.toml $out/etc/liana/config.toml
   '';
 
-  desktopItems = [
+  desktopItems = if withGui then [
     (makeDesktopItem {
       name = "Liana";
       exec = "liana-gui";
       icon = "liana";
       desktopName = "Liana";
       comment = meta.description;
     })
-  ];
+  ] else null;
 
-  doCheck = true;
+  doCheck = if withGui then true else false;
```

Comment:
Is something failing for `lianad`?

Reply:
yes, if I revert this back to `doCheck = true;` and do `nix-build -A lianad`:

```
failures:

---- config::tests::config_directory stdout ----
thread 'config::tests::config_directory' panicked at src/config.rs:482:13:
assertion failed: filepath.as_path().starts_with("/home/")
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    config::tests::config_directory

test result: FAILED. 50 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 4.07s

error: test failed, to rerun pass `--lib`
error: builder for '/nix/store/78if09d6hy9sr6ymcw6m1c6xzdh0si4z-lianad-5.0.drv' failed with exit code 101;
       last 10 log lines:
       > assertion failed: filepath.as_path().starts_with("/home/")
       > note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
       >
       >
       > failures:
       >     config::tests::config_directory
       >
       > test result: FAILED. 50 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 4.07s
       >
       > error: test failed, to rerun pass `--lib`
       For full logs, run 'nix log /nix/store/78if09d6hy9sr6ymcw6m1c6xzdh0si4z-lianad-5.0.drv'.

```

Reply:
based on my understand of what you said here https://github.com/fort-nix/nix-bitcoin/pull/708#issuecomment-2199512798

it seems we should move it out of `by-name`?

Reply:
Ok we can leave it then :+1: